src/backend/joanie/core/models/products.py

```diff
@@ -720,20 +725,27 @@ def clean(self):
             )
             raise ValidationError({"__all__": [message]})
 
-        # The user should not be enrolled in another opened course run of the same course.
-        if (
-            self.created_on is None
-            and self.user.enrollments.filter(
+        if self.created_on is None:
+            # The user should not be enrolled in another opened course run of the same course.
+            if self.user.enrollments.filter(
                 course_run__course=self.course_run.course,
                 course_run__end__gte=timezone.now(),
                 is_active=True,
-            ).exists()
-        ):
-            message = _(
-                "You are already enrolled to an opened course run "
-                f'for the course "{self.course_run.course.title}".'
-            )
-            raise ValidationError({"user": [message]})
+            ).exists():
+                message = _(
+                    "You are already enrolled to an opened course run "
+                    f'for the course "{self.course_run.course.title}".'
+                )
+                raise ValidationError({"user": [message]})
+
+            # Enforce was_created_by_order to True if the enrollment relies on
+            # a not listed CourseRun
+            if not self.course_run.is_listed:
+                self.was_created_by_order = True
+            # Enforce was_created_by_order to False if the enrollment relies on
+            # a listed CourseRun linked to any product
+            elif not self.course_run.course.targeted_by_products.exists():
+                self.was_created_by_order = False
```

Comment:
since the field is False by default, I would just add after existing line #752 below:
self.was_created_by_order = True

Reply:
TODO Write a comment to explain the use case

Reply:
Instead raise a `ValidationError` if the `was_created_by_order` flag contains a wrong value.

Reply:
@sampaccoud I push a fixup commit. Now the clean method raise an error when the `was_created_by_order` is not set properly according to the context.